ui/src/app/job-list/table/table.component.ts

```diff
@@ -249,4 +264,56 @@ export class JobsTableComponent implements OnInit {
       }
     }
   }
+
+  openbulkEditDialog(): void {
+    // get default values for bulk edit fields in dialog
+    for (let bulkFieldItem of this.bulkLabelFields) {
+      const label = bulkFieldItem.displayField.field.replace('labels.', '');
+      bulkFieldItem.default = this.selection.selected[0].labels[label];
+      for (let job of this.selection.selected) {
+        const jobLabelValue = job.labels[label] || '';
+        if (bulkFieldItem.default !== jobLabelValue) {
+          bulkFieldItem.default = null;
+        }
+      }
+    }
+
+    let dialogRef = this.bulkEditDialog.open(JobsBulkEditComponent, {
+      disableClose: true,
+      data: {
+        'bulkLabelFields' : this.bulkLabelFields,
+        'selectedJobs' :  this.selection.selected,
+        'newValues': []
+      }
+    });
+
+    dialogRef.afterClosed().subscribe((result) => {
+      if (!result || !this.doesResultHaveLabelChanges(result)) {
+        return;
+      }
+      let lastJobUpdated: QueryJobsResult[];
+      for (let job of result.jobs) {
+        this.jobManagerService.updateJobLabels(job.id,
+            this.prepareUpdateJobLabelsRequest(result.fields))
+          .then((response: UpdateJobLabelsResponse) => {
+            job.labels = response.labels;
+            lastJobUpdated = job;
+          })
+          .catch((error) => this.handleError(error));
+      }
+      this.onJobsChanged.emit(lastJobUpdated);
```

Comment:
At this point in the code, we just sent a bunch of async XHRs to update job labels, but they haven't returned yet. We should only be emitting a change event once the calls actually complete.

Accumulating the Promises via `Promise.all()` may be what you're looking for.